denis.roman/1.rb

```diff
@@ -0,0 +1,14 @@
+require 'faraday'
+
+class Image
+  def self.download(url)
+    response = Faraday.get url
+    extension = response.headers['content-type'].split('/').last
+    content_type = response.headers['content-type'].split('/').first
+    File.open("download.#{extension}", 'w') { |fp| fp.write(response.body) }
```

Comment:
You need to raise exception before saving the file. What is the point raising an exception when you already saved the file?